amethyst_ui/src/widgets.rs

```diff
@@ -187,7 +187,10 @@ macro_rules! define_widget {
         /// A $t widget, containing references to its associated entities.
         #[derive(Debug, Clone)]
         pub struct $t {
-            $($field: $crate::Entity),*
+            $(
+                /// `$field` Entity
```

Comment:
I don't think macros replace anything in doc comments. So this might look weird in the API docs

Reply:
Ah, I was not aware of that. In that case, the attribute variant could be used here:
```rust
#[doc="A $t widget, containing references to its associated entities."]
pub struct $t { ... }
```

Reply:
Unfortunately, it doesn't work in strings either :(

Reply:
I've tried before, `stringify!` also doesn't work here. So the only way I know of is to just put docs inside the macro invocation and to-expand them using the macro.

Reply:
Ah, that sucks... But since all generated method do roughly the same thing, maybe the documentation doesn't need to be too detailed. This probably warrants another look, but should not be a blocker for this PR.

Reply:
one day in the future i may turn that into a proc macro